File: solutions/exercise-3/src/main/java/io/miragon/training/adapter/outbound/cibseven/MembershipProcessAdapter.java
package io.miragon.training.adapter.outbound.cibseven;

import io.miragon.training.application.port.outbound.MembershipProcess;
import io.miragon.training.domain.Membership;
import org.cibseven.bpm.engine.RuntimeService;
import org.springframework.stereotype.Component;

import java.util.Map;

@Component
public class MembershipProcessAdapter implements MembershipProcess {

    private final RuntimeService runtimeService;

    public MembershipProcessAdapter(RuntimeService runtimeService) {
        this.runtimeService = runtimeService;
    }

    @Override
    public void startProcess(Membership membership) {
        runtimeService.startProcessInstanceByKey("subscribeNewsletter", Map.of(
                "membershipId", membership.id().value().toString(),
                "email", membership.email().value(),
                "name", membership.name().value(),
                "age", membership.age().value()
        ));
    }
}

File: exercises/src/main/kotlin/io/miragon/training/adapter/outbound/cibseven/MembershipProcessAdapter.kt
package io.miragon.training.adapter.outbound.cibseven

import io.miragon.training.adapter.process.MembershipProcessApi
import io.miragon.training.application.port.outbound.MembershipProcess
import io.miragon.training.domain.Membership
import org.cibseven.bpm.engine.RuntimeService
import org.springframework.stereotype.Component

@Component
class MembershipProcessAdapter(
    private val runtimeService: RuntimeService,
) : MembershipProcess {

    override fun startProcess(membership: Membership) {
        // TODO Aufgabe 1:
        //  Starte eine neue Prozessinstanz mit dem Key aus MembershipProcessApi.PROCESS_KEY.
        //  Übergib folgende Prozess-Variablen als Map:
        //    - "membershipId" -> membership.id.value.toString()
        //    - "email"        -> membership.email.value
        //    - "name"         -> membership.name.value
        //    - "age"          -> membership.age.value
        //  Hinweis: runtimeService.startProcessInstanceByKey(key, variables)
        TODO("Aufgabe 1: Starte den Prozess via RuntimeService")
    }
}
